go/vt/topo/etcd2topo/watch.go

```diff
@@ -62,15 +63,35 @@ func (s *Server) Watch(ctx context.Context, filePath string) (*topo.WatchData, <
 	go func() {
 		defer close(notifications)
 
+		var count int
 		for {
 			select {
+
 			case <-watchCtx.Done():
 				// This includes context cancelation errors.
 				notifications <- &topo.WatchData{
 					Err: convertError(watchCtx.Err()),
 				}
 				return
-			case wresp := <-watcher:
+			case wresp, ok := <-watcher:
+				if !ok {
+					if count > 10 {
+						time.Sleep(time.Duration(count) * time.Second)
+					}
+					count++
+					cur, err := s.cli.Get(ctx, nodePath)
+					if err != nil {
+						continue
+					}
```

Comment:
Before watching, we need to handle the case where the file disappeared, something like:
if len(cur.Kvs) != 1 {
  // Node doesn't exist.
  notifications <- &topo.WatchData{
    Err: topo.ErrNoNode,
  }
  return
}